views/partials/dashboards/preferences.hbs

```diff
@@ -1,47 +1,63 @@
 {{#getUserPreferences}}
-  <section class="container clear-header preferences sr" data-page-label="Preferences">
-    <div class="row jst-cntr no-vertical-padding">
-      <div class="pref">
-        <h2 class="pref-headline">{{ getString "monitor-preferences" }}</span></h2>
-      </div>
+  <section class="clear-header preferences sr jst-cntr flx" data-page-label="Preferences">
+    <div class="pref">
+      <h2 class="pref-headline">{{ getString "monitor-preferences" }}</h2>
     </div>
   </section>
-  <section>
-    <div class="row flx-col">
-      <div class="pref">
-        <h3 class="pref-section-headline section-headline txt-purple7">{{ getString "breach-alert-notifications" }}</h3>
-        <section>
-          {{#each communicationOptions}}
-            <label class="radio-container">
-              <input class="radio-comm-option" data-comm-option="{{ optionId }}" data-form-action="update-comm-option" data-csrf-token="{{ ./../../csrfToken }}" type="radio" {{ optionChecked }} name="1">
-              <span class="radio-label">{{{ labelString }}}</span>
-              <span class="checkmark"></span>
-            </label>
-          {{/each}}
-        </section>
-      </div>
-      <div class="pref">
-        <h3 class="pref-section-headline section-headline txt-purple7">{{ getString "email-addresses-title" }}</h3>
-        {{#each emails}}
-          {{#ifCompare email_addresses.length ">" 0}}
-            <h4 class="email-pref {{ this.className }}">{{ this.subhead }}</h4>
-            {{#each this.email_addresses}}
-              {{> email-card prefs=./../../../preferences csrfToken=./../../csrfToken}}
-            {{/each}}
-          {{/ifCompare}}
+  <section class="flx flx-col">
+  
+    <!-- Communication Preferences -->
+    <div class="pref">
+      <h3 class="pref-section-headline section-headline txt-purple7">{{ getString "breach-alert-notifications" }}</h3>
+      <div>
+        {{#each communicationOptions}}
+          <label class="radio-container">
+            <input class="radio-comm-option" data-comm-option="{{ optionId }}" data-form-action="update-comm-option" data-csrf-token="{{ ./../../csrfToken }}" type="radio" {{ optionChecked }} name="1">
+            <span class="radio-label overflow-break">{{{ labelString }}}</span>
```

Comment:
Re-checked that this `labelString` value does not come from user-generated content. Although it's user-translated content.

@flodolo I assume there are checks in Pontoon (either automatic or manual) that attackers couldn't sneak XSS markup into their translations?

Reply:
Does this string still use embedded HTML?

```diff
- labelString: LocaleUtils.fluentFormat(locales, "to-affected-email", {primaryEmail: `<span class="bold">${primaryEmail.email}</span>`}),
+ labelString: LocaleUtils.fluentFormat(locales, "to-affected-email"),
```

Not sure if this is the same `labelString` as template-helpers/dashboard.js:102, which seemingly got rid of the `<span>` and `primaryEmail.email` address.

Or if we need some fancy HTML sanitizer filters so if we _have_ to use triple `{{{` in the templates, we can at least limit which HTML tags we want to support per string.
Like some mythical `{{{ labelString | allowedTags=["span"] }}}`.

Reply:
Current instances of `{{{...}}}` in the codebase:

```sh
git grep -EIno "\{\{\{.*?\}\}\}"

views/layouts/default.hbs:26:{{{ body }}}
views/layouts/default_email.hbs:110:{{{ getEmailHeader }}}
views/layouts/email_l10n_mockups.hbs:30:{{{ getEmailHeader }}}
views/monitor.hbs:6:{{{ headline }}}
views/monitor.hbs:35:{{{ getString "latest-breach" brandName="<span class='fxm'>Firefox Monitor</span>" }}}
views/partials/breach-detail-content-group.hbs:9:{{{ this.copy }}}
views/partials/dashboards/breaches-dash.hbs:5:{{{ welcomeMessage }}}
views/partials/dashboards/breaches-dash.hbs:50:{{{ this }}}
views/partials/dashboards/preferences.hbs:17:{{{ labelString }}}
views/partials/email-card.hbs:23:{{{ fluentNestedBold "appears-in-x-breaches" breachCount=breaches.length }}}
views/partials/email_partials/email_disclaimers.hbs:9:{{{ getString "learn-more-about-fxm" fxmLink='<a style="color: #0060df; font-weight: 400;" href="https://monitor.firefox.com/?utm_source=fx-monitor&utm_medium=email&utm_campaign=learn-more-link" target="_blank" rel="noopener noreferrer">Firefox Monitor</a>'}}}
views/partials/email_partials/email_disclaimers.hbs:10:{{{ getString "hibp-attribution" hibp-link='<a id="hibp-link" style="color: #0060df; font-weight: 400;" href="https://www.haveibeenpwned.com" target="_blank" rel="noopener noreferrer">Have I Been Pwned</a>' }}}
views/partials/email_partials/email_faqs.hbs:7:{{{ makePreFxaSubscriberMessage }}}
views/partials/email_partials/email_footer.hbs:5:{{{ getEmailFooterCopy }}}
views/partials/email_partials/pre-fxa.hbs:5:{{{ getEmailHeader }}}
views/partials/email_partials/pre-fxa.hbs:27:{{{ paragraph }}}
views/partials/email_partials/report.hbs:40:{{{ getBreachSummaryHeadline }}}
views/partials/header/fxa-menu.hbs:8:{{{ getSignedInAs }}}
views/partials/hibp-attribution.hbs:2:{{{ getString "hibp-attribution" hibp-link='<a id="hibp-link" class="text-link medium" href="https://www.haveibeenpwned.com" data-ga-link="" data-event-category="Utility & Outbound Links" data-event-label="Clicked-HIBP" target="_blank" rel="noopener noreferrer">Have I Been Pwned</a>' }}}
views/partials/subpages/confirm.hbs:2:{{{ makeEmailAddedToSubscriptionString email }}}
views/partials/subpages/confirm.hbs:3:{{{ makeEmailVerifiedString }}}
views/partials/subpages/unsubscribe.hbs:2:{{{ getString "unsub-headline" }}}
views/partials/subpages/unsubscribe.hbs:3:{{{ getString "unsub-blurb" }}}
views/partials/top-level/security-tips.hbs:27:{{{ this }}}
views/partials/top-level/security-tips.hbs:55:{{{ subhead }}}
views/partials/top-level/security-tips.hbs:67:{{{ subhead }}}
views/partials/top-level/security-tips/dos-and-donts.hbs:20:{{{ this.do }}}
views/partials/top-level/security-tips/security-tip-callout.hbs:6:{{{ securityTip.tipHeadline }}}
views/partials/top-level/security-tips/security-tip-callout.hbs:9:{{{ securityTip.tipSubhead }}}
views/partials/top-level/security-tips/tip-content.hbs:3:{{{ this }}}
views/scan.hbs:10:{{{ this.headline }}}
views/scan.hbs:12:{{{ this.subhead }}}
views/scan.hbs:16:{{{ fluentNestedBold "scan-results-known-breaches" breachCount=foundBreaches.length }}}
views/subpage.hbs:5:{{{ headline }}}
views/subpage.hbs:6:{{{ subhead }}}
```

Reply:
The second option still has a span. https://github.com/mozilla/blurts-server/blob/1f5039cdd5f7c54a4ab0951fd629773dee65f2af/template-helpers/dashboard.js#L108